### ISSUE TITLE
postcss-loader warnings

### ISSUE THREAD

**Issue by egoist** [vuejs/vuepress#278]
https://github.com/vuejs/vuepress/blob/2d6f02f3b06ffd58d1f0000001a621545229d458/lib/build.js#L99-L107

VuePress didn't handle webpack warnings, but if you add following lines:

```js
if (stats.hasWarnings()) {
  return reject(new Error(stats.toString()))
}
```

And if you run `yarn build --debug` in VuePress repo, there're actually postcss-loader warnings:

<img width="953" alt="2018-04-27 10 04 08" src="https://user-images.githubusercontent.com/8784712/39366600-f755b864-4a66-11e8-8423-3a1e6395c06f.png">

Not sure if this is a vue-loader bug since I can only reproduce it in this repo. I've tried console.log `this.sourceMap` in `vue-loader`, seems it's always `true` even if I already set `webpackConfig.devtool: false`

Test against latest commit: https://github.com/vuejs/vuepress/commit/2d6f02f3b06ffd58d1f0000001a621545229d458

**Comment by meteorlxy** [comment]
And another warning:

`DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead`

**Comment by ulivz** [comment]
Reproduced this problem, It works when set `sourceMap` to `true`. but what is strange is that no source map was generated. maybe need more investigation. 

```js
rule.use('postcss-loader').loader('postcss-loader').options(Object.assign({
        plugins: [require('autoprefixer')],
        sourceMap: true
      }, siteConfig.postcss))
```

**Comment by meteorlxy** [comment]
```js
  if (debug) {
    // config.devtool('source-map')
  } else if (!isProd) {
    config.devtool('cheap-module-eval-source-map')
  }
```

I comment that line and the warning disappear. Maybe the `sourceMap` should be set to `debug`?

**Comment by ulivz** [comment]
Maybe?

```js
  if (!isProd) {
    config.devtool('cheap-module-eval-source-map')
  } else if (debug) {
    config.devtool('source-map')
  }
```

**Comment by meteorlxy** [comment]
emmmm... `dev` command doesn't have `--debug` mode, it only works in `build` command, in which `isProd` is always `true`

**Comment by meteorlxy** [comment]
Oh my god, me neither!!

**Comment by ulivz** [comment]
So amazing, what happened?

**Comment by meteorlxy** [comment]
I have  nooooooooo idea.

**Comment by ulivz** [comment]
Welcome to find it out, thanks!

**Comment by ulivz** [comment]
（老哥，咱俩是蠢到一块去了 😂）

**Comment by egoist** [comment]
@yyx990803 might be a bug in vue-loader?

**Comment by meteorlxy** [comment]
Two things cause these warnings:

1. `stylus-loader` default `sourceMap` options is `{ comment: false }` but not `false`, so we have to disable it mannually in `production` mode. See [related code of stylus-loader](https://github.com/shama/stylus-loader/blob/5258d97e2ae2b1bcdfbc1b382afbff937ee656d9/index.js#L46-L48)

2. `css-loader` with `module: true` **may** always have `sourceMap` before `postcss-loader`. (I'm not sure about this now).

After I fix the `stylus-loader`'s `sourceMap` problem, there are still two warnings about file `docs/guide/using-vue.md` and `docs/zh/guide/using-vue.md`, in which there is a `css-module` usage at the end of those two pages.

I will make a PR to fix this issue later.

**Comment by ulivz** [comment]
Fixed at 34c7f9936c1122f3e0c1b7329fbf26bcac5da1d9
Thanks to Evan at https://github.com/postcss/postcss-loader/pull/361
